packages/dropdown/src/react/item.tsx

```diff
@@ -30,28 +39,44 @@ interface DropdownItemProps
 export const Item = forwardRef<HTMLButtonElement, DropdownItemProps>(
   ({ disabled, onClick, value, icon, children, ...rest }, forwardedRef) => {
     const context = useContext(DropdownContext)
+    const ref = useRef<HTMLButtonElement>()
+    useImperativeHandle(
+      forwardedRef,
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      () => ref.current as HTMLButtonElement
+    )
+    const [active, setActive] = useState(false)
+    const isActive = useCallback(() => {
+      ref.current && setActive(document.activeElement === ref.current)
+    }, [ref, setActive])
+    useEffect(() => {
+      isActive()
+    }, [isActive])
     const valueExists = typeof value !== 'undefined'
-    const isActive =
-      (valueExists && context.activeItem?.value === value) ||
-      context.activeItem?.label === children
     const isSelected =
       (valueExists && context.selectedItem?.value === value) ||
       context.selectedItem?.label === children
 
-    const handleClick = (evt: React.MouseEvent) => {
+    const handleClick = (evt: React.SyntheticEvent) => {
```

Comment:
is this always going to me a MouseEvent? if so i'd suggest using MouseEvent or MouseEventHandler so you can avoid casting the events below

Reply:
I also use this same function for keydown `Enter`. Could simply make a copy but for keydown was reducing lines.

Reply:
Have done a union type in other code, eg:

```
React.MouseEvent | React.KeyboardEvent
```

And then done type narrow (no cast) in the body as needed.

Reply:
I'll give that a shot just to be clear this an internal function you know? Like how important is this so long as the onClick passed to the component has the signature we desire?